xhu.py

```diff
@@ -65,12 +76,45 @@ def get_info(path: str, root: pathlib.Path) -> typing.Tuple[
         path,
         pathlib.Path(app.config["DATA_ROOT"]),
     )
-
     data_file, metadata_file = get_paths(dest_path)
-
     return data_file, load_metadata(metadata_file)
 
 
+def apply_quota(root: pathlib.Path, quota: int):
+    """ Get the files, sorted by last modification date and the sum of their
+        sizes.
+    """
+    if not quota:
+        return
+
+    file_list = []
+    total_size = 0
+    # We assume a file structure whereby files are are stored inside
+    # uuid() directories inside the root dir and that there aren't any files in
+    # the root dir itself.
+    for uuid_dir in os.listdir(root):
+        for path, dirnames, filenames in os.walk(root/uuid_dir):
+            for name in [n for n in filenames if n.endswith('.data')]:
+                fp = os.path.join(path, name)
+                size = os.path.getsize(fp)
+                total_size += size
+                modified = os.stat(fp)[stat.ST_MTIME]
+                file_list.append((modified, path, name, size))
+
```

Comment:
From my reading, this sums the files of all users. 

Also this is O(n) in the number of stored files, which I don’t like at all. This must be condensed into an O(1) operation.

See the main review comment for details.